packages/parameters/src/BaseProvider.ts

```diff
@@ -49,6 +55,11 @@ abstract class BaseProvider implements BaseProviderInterface {
 
     let value;
     try {
+      // Type assertion is needed because the SSM provider has a different signature for the get method
+      if (options && isSSMGetOptionsInterface(options) && options.decrypt) {
+        options.sdkOptions = options.sdkOptions || {};
+        options.sdkOptions.WithDecryption = true;
+      }
```

Comment:
Would it be better to move this into `_get()` of the `SSMProvider` class so that there's no implementation specific logic in the base class? This comment also applies to `getMultiple()`.

Reply:
Hi @buggy thanks for the review. 

That's a good point, I'll look into it and see if I can bring that logic into `SSMProvider`.

Reply:
I have applied the change you suggested in this commit [e47ddf67](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/1187/commits/e47ddf67f14cde4e41c5bc2dd7a71eedd990d549). 

Thank you!